### PR TITLE
Feature/staging environment deploy

### PR DESCRIPTION
**Pull request by janbaykara**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I've checked the spec (e.g. Figma file) and documented any divergences.
- [ ] My code follows the code style of this project.
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I've updated the documentation accordingly.
- [ ] Replace unused checkboxes with bullet points.

### CODE REVIEW

**Review by lowlandjuju — APPROVED** [review]
This looks great

File: Dockerfile [hunk 1]
```diff
@@ -6,8 +6,13 @@ RUN SKIP_MIGRATE=1 bash install.sh
...
-  NODE_ENV=production
+COPY --chown=app:app . ./
+ENV DJANGO_SETTINGS_MODULE=smartforests.settings.production
+ENV NODE_ENV=production
 
+# Build
 RUN pipenv run bash .bin/build.sh
+
+# Run
+EXPOSE ${PORT:-8080}
```
Comment (lowlandjuju) [inline Dockerfile:17]: Is this line actually necessary? Given the port entry in the fly.toml?

File: Dockerfile [hunk 2]
```diff
@@ -15,4 +15,4 @@ RUN pipenv run bash .bin/build.sh
 
 # Run
 EXPOSE ${PORT:-8080}
-#RUN pipenv run bash .bin/release.sh
\ No newline at end of file
+RUN pipenv run bash .bin/release.sh
```
Comment (lowlandjuju) [inline Dockerfile:18]: The reason for the [processes] section is that `fly secrets` are not available at build time, so if `.bin/run.sh` is in the Dockerfile, then it won't be able to access the env vars

File: Dockerfile (hunk 1 above)
Comment (janbaykara) [reply Dockerfile:17]: I think it is, there needs to be a match up between Fly and Docker.

File: Dockerfile (hunk 2 above)
Comment (janbaykara) [reply Dockerfile]: [You're totally right, apologies](https://fly.io/docs/reference/build-secrets/):

> You can set [secrets](https://fly.io/docs/reference/secrets/) for your applications, but these are only available at run-time. They aren't available when building your Docker image without a little extra work.

@lowlandjuju what secrets were required at build time?

**Pull request merged**